### ISSUE TITLE
Add configuration to no-inferrable-types to ignore function return types

### ISSUE THREAD

**Issue by devison** [palantir/tslint#2737]
The `no-inferrable-types` currently complains if a function has a return type annotation when it could be inferred.

#### TypeScript code being linted

```ts
    function calcSum(x: number, y: number): number {
        return x + y;
    }
```

_This is simple example - in more complex functions it can take a bit of inference to figure out what the return type is._

with `tslint.json` configuration:

```json
"no-inferrable-types": true
```

#### Actual behavior

```
ERROR: Type number trivially inferred from a number literal, remove type annotation
```
#### Expected behavior

It would be nice if the rule had a configuration to skip function return types.

#### Proposed solution

Ability to specify this in `tslint.json`:

```
"no-inferrable-types": [true, "ignore-return"]
```

**Comment by devison** [comment]
woops I was mistaken.
